### DISCUSSION TITLE
Doing simple stuff (for a genealogy tree) is overcomplicated. Let's use the tree-view as a simple way to interact with the family tree!

### DISCUSSION THREAD

**Discussion by lucaberti** [gramps-project/gramps-web#410] [Ideas]
Gramps Web is a great tool for hobby genealogists who don't want to spent a lot of money on overpiced genealogy tools. Nevertheless it is so frustrating when it comes to do most basic things a genealogy tree should do. For instance: my cousine got married and got a son. In order to update the tree: 
1. I had to manually add the husband as a new person
2. I had to add the family wich relates both person
3. I had to add their son as a new person (it is impossible to add new elements when it comes to the dropdown menu).
4. I add to add thier son to the family

This is really overcomplicatated for such a simple task.

So... would not ben more easy to be able to add/manage people to the tree, directly from the tree trough a right-click menu? In my humble opinion the tree-view should become the main navigation tool, and the preferred work environtment: it is the most user-friendly way.

**Comment by DavidMStraub** [comment]
Thanks for starting the discussion, it's a needed one.

Some random thoughts:

- I don't see why Gramps Web is just for hobby genealogists
- I don't even see the pricing (or lack thereof) as the main advantage, but control, data sovereignty, open source
- I agree the UX for adding objects is bad and there is lots of potential to improve it
- Being able to add people from the tree view would indeed be nice 

But:

- I don't think adding this feature would solve the UX issues - adding places, sources, events, ... would still be the same
- I don't like right click that much because it doesn't translate easily to a pattern on mobile
- We have to make sure we don't introduce a pattern that encourages adding people without sources (so, adding a source should be easy also in the edit tree view)

One possibility would be to not actually use the tree/graph view, but display a little tree on a person's profile in edit mode like this on FamilySearch:

![image](https://github.com/gramps-project/gramps-web/assets/10965193/4b28c99f-d7c8-4aa6-97f2-8efa8a8de09d)

**Reply by lucaberti** [reply]
Sorry, I did not wanted to diminish Gramps: it is a great software that I really like and that I'd like to see evolve in its full potential. And you cited most of the reasons why: data control, open sourse, self hosting and (last and least ;)) price.

**Comment by geostag** [comment]
I appreciate this discussion! I agree totally with @DavidMStraub that there are a lot of good reasons to use gramps, that might even be good reasons for professional genealogists. And that gramps has a very straight forward UX from an expert point of view, who is very familiar with the underlying data model. However, it is neither straight forward for people, who are not interested in the data model nor is it addressing real-world use cases like the mentioned one.

I would love it, if we could address this on a more generic way than to just add some short cuts.

I did some search on the net concerning UX in genealogy software, and there is very very little out there ([1](https://www.figma.com/community/file/1221130545200803501/famtree-ux-case-study-template-community), [2](https://www.behance.net/gallery/78173379/UX-for-Building-a-family-tree-App)). Even the large commercial sites (family search, ancestry, myheritage) do have more or less complicated interfaces when it comes to editing data and linking persons.

Getting good UX for genealogy software seems to be a hard task. Referring to Nielsen Norman Group, this kind of software is a [complex one](https://www.nngroup.com/articles/complex-application-design-framework/). Which means we might need real UX experts to create user-friendly, robust, failsave, pretty UX, rater than trying to fix it on isolated use-cases.

**Comment by lucaberti** [comment]
I fully agree with you both that the UX should be carefully tought of in a generic and organic way. I did my thesis on Human Computer Interaction, so I have some basics on that, but I feel quite rusty at it.

This is surelly a quite complex UX to come up with. Nevertheless I shuld say that MyHerigate is quite good for "normal" tasks. Mori ([web](https://www.mori.co/)) look like clean and very usable. 

Generally speaking, I think that the UX main principle is that every main function should be accessible from the main tree view.
